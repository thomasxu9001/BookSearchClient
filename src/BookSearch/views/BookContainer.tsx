import React, {Component} from 'react';
import {withRouter, RouteComponentProps, } from 'react-router';
import {Link} from 'react-router-dom';

import BookDetail from "../components/BookDetail";
interface StateProps {

}

interface State {
}


export type Props = StateProps & RouteComponentProps<{ id: string }>;

export class BookContainer extends Component<Props, State> {

    render() {
        const {match} = this.props;

        return (
            <div className="bookContainer">
                <Link className="backToList" to={'/'}>
                    Back to Book List
                </Link>
                <BookDetail bookId={+match.params.id}/>
            </div>
        );
    }
}

export default withRouter(BookContainer);



