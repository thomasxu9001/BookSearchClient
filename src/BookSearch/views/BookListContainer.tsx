import React, {Component} from 'react';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';

import BookCard from '../components/BookCard';
import {Book} from '../Book';
import "../Book.less";
import SearchInput from '../components/SearchInput';
import PaginationFooter from '../../PaginationFooter';



interface StateProps {

}

interface State {
    error: any
    isLoaded: boolean
    items: []
    totalPage: number
    currentPage: number
    filter: string
}


export type Props = StateProps;

export class BookListContainer extends Component<Props, State> {

    constructor(props: Props) {
        super(props);
        this.state = {
            error: null,
            isLoaded: false,
            items: [],
            totalPage: 0,
            currentPage: 1,
            filter:''
        };
    }

    componentDidMount() {
        this.searchBook();
    };

    searchBook = (keyword?: string, page?: number) => {
        const baseUrl = "http://localhost:8000/Rest/books";
        let url = keyword ? baseUrl + '?search=' + keyword + '&': baseUrl + '?';
        url = page ? url + 'page=' + page : url;

        fetch(url, {
            method: 'GET'
        })
            .then(res => res.json())
            .then(
                (result) => {

                    this.setState({
                        filter: keyword ? keyword : '',
                        isLoaded: true,
                        items: result['items'],
                        totalPage: result['total_page'],
                        currentPage: result['page']
                    });
                },
                // Note: it's important to handle errors here
                // instead of a catch() block so that we don't swallow
                // exceptions from actual bugs in components.
                (error) => {
                    this.setState({
                        filter: keyword ? keyword : '',
                        isLoaded: true,
                        error,
                        totalPage: 0,
                        currentPage: 1
                    });
                }
            );
    };

    onPaginationClickHandler = ( page: number) => {
        const {filter} = this.state;
        this.searchBook(filter, page);
    };

    // Generate BookCard list
    getBookList = () => {
        const {items} = this.state;
       if (items && items.length > 0){
           return items.map((item: Book) => {
                   return <BookCard key={item.id} item={item}/>
               }
           );
       }
        return <div className="noResult">No results found.</div>
    };

    render() {
        const {isLoaded, totalPage, currentPage} = this.state;

        return (
            <>
                <div className='bookMainHeader'>
                    <h1><FontAwesomeIcon icon="book"/> Smart Book Search</h1>
                    <SearchInput searchBook={this.searchBook}/>
                </div>
                {isLoaded ? (
                    <div className="bookList">
                        {this.getBookList()}
                    </div>
                ) : (
                    `Loading......`
                )}
                <PaginationFooter totalPage={totalPage} currentPage={currentPage} onPageChange={this.onPaginationClickHandler}/>
            </>
        );
    }
}

export default BookListContainer;



