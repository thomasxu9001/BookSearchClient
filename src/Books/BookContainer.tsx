import React, {Component} from 'react';
import BookDetail from "./BookDetail";
interface StateProps {

}

interface State {
}


export type Props = StateProps;

export class BookContainer extends Component<Props, State> {

    render() {

        return (
            <div>
                <div> Book Container Page</div>
                <BookDetail/>
            </div>
        );
    }
}
export default BookContainer;



