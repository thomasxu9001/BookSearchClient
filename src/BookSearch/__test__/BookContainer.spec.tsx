import React from 'react';
import {createLocation, createMemoryHistory} from 'history';
import {match as routerMatch} from 'react-router';
import {BrowserRouter} from 'react-router-dom'

import {mount, ReactWrapper} from 'enzyme';
import {BookContainer, Props} from "../views/BookContainer";

const history = createMemoryHistory();
const path = `/route/:id`;

const match: routerMatch<{ id: string }> = {
    isExact: false,
    path,
    url: path.replace(':id', '1'),
    params: {id: "1"}
};

const location = createLocation(match.url);

describe('BookContainer', () => {

    let
        subject: ReactWrapper<Props>
    ;

    beforeEach(() => {

        subject = mount(
            <BrowserRouter>
                <BookContainer
                    history={history}
                    location={location}
                    match={match}
                />
            </BrowserRouter>
        );
    });

    it('looks good', () => {
        expect(subject).toMatchSnapshot();
    });

});
