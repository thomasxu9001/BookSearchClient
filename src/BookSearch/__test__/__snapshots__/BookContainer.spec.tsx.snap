// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookContainer looks good 1`] = `ReactWrapper {}`;
